script.js

```diff
@@ -1 +1,318 @@
-// Please implement exercise logic here
+// completed => Match Game setTimeout / Match Game Match /Full Match Game /Match Game Timer
+
+// boardSize has to be an even number
+const boardSize = 4;
+const board = [];
+let firstCard = null;
+let firstCardElement;
+let deck;
+let state;
+let timer;
+let firstClick = true;
+const milliseconds = 1;
+const delayInMilliseconds = 1;
+let submitBtn;
+let inputBox;
+let count = 0;
+function sleep(ms) {
```

Comment:
We might want to prevent the user from clicking using the sleep time as well to prevent another `squareClick` from running